ClientApp/viking-vault/src/UserDashboard/DownloadTypeForm.tsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { Redirect } from 'react-router-dom';
+import downloadIcon from '../Resources/images/download-icon.png'
+import { constants } from "../Resources/Constants";
+
+const API_URL = `${constants.baseUrl}pdfgenerator`;
+const LOGIN_ROUTE ="/login";
+
+class DownloadDropdown extends React.Component {
+    state = {
+      isOpen: false,
+      timeFilter: "",
+      redirect: false
+    };
+  
+    toggleOpen = () => this.setState({ isOpen: !this.state.isOpen });
+  
+    generatePDF = () => {
+        let token = sessionStorage.getItem("Authentication-Token");
+        if(token === null){
+            this.setState({
+                redirect: true
+            })
+        }
+        else {
+            var pdfURL = API_URL + "/"+ this.state.timeFilter;
+            fetch(pdfURL, {
+                    method: "GET",
+                    headers: {
+                        "x-access-token":  token.toString()
+                    }
+                })
+                .then((response) => response.blob())
+                .then((blob) => {
+                    if(blob != null) {
+                    const url = window.URL.createObjectURL(new Blob([blob]));
+                    const link = document.createElement('a');
+                    link.href = url;
+                    link.setAttribute('download', `Viking Vault Report.pdf`);
+                    document.body.appendChild(link);
```

Comment:
Hmm... why do you try to manipulate the DOM directly instead of doing it through the React library?

Reply:
I deleted this line
**document.body.appendChild(link);**

Copy-paste... Hihi... Thanks!